include/openssl/kdf.h

```diff
@@ -83,6 +83,29 @@ OPENSSL_EXPORT int EVP_PKEY_CTX_add1_hkdf_info(EVP_PKEY_CTX *ctx,
                                                const uint8_t *info,
                                                size_t info_len);
 
+// SSH-specific KDF
+//
+// This KDF should only be called from SSH client/server code; it's not a
```

Comment:
Does exposing this KDF function via `EVP_KDF_XYZ` APIs violate FIPS?

Reply:
FIPS doesn't care about APIs, just proper crypto (and evidence of that). SSH KDF isn't available via the `EVP` APIs, just directly as `SSHKDF()`.

Reply:
https://www.openssl.org/docs/man3.0/man7/EVP_KDF-SSHKDF.html

Reply:
Surely. I was trying to find a good spot for it without creating a new file; IMHO we should put all the KDFs together in one logical spot, but that moves us away from Boring/Open.

Reply:
@nebeid Thanks for the pointer. I was under the assumption that AWS-LC's API is suppose to be 100% compatible with OpenSSL so it can be a drop-in replacement for OpenSSL, which is not true. Now that AWS-LC is providing some APIs that are not in OpenSSL (and possibly not in BoringSSL), it's better to have their corresponding header files in a separate directory like `include/aws-lc` rather than `include/openssl`. In this way, consumers of AWS-LC can easily distinguish APIs that are specific to AWS-LC that are not supported in OpenSSL.